packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,173 @@
+  onTableHeaderClick: (payload: keyof LocationAccess<Permission>) => void;
+  sortState: SortState<LocationAccess<Permission>>;
+}) => {
+  const { selection, direction } = sortState;
+
+  const columns = [
+    {
+      children: (
+        <ButtonElement
+          variant="sort"
+          className={TABLE_HEADER_BUTTON_CLASS_NAME}
+        >
+          Name
+          <IconElement
+            variant={
+              selection === 'scope' && direction !== 'none'
+                ? `sort-${direction}`
+                : 'sort-indeterminate'
+            }
+          />
+        </ButtonElement>
+      ),
+      key: 'th-scope',
+      className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--scope`,
+      onClick: () => onTableHeaderClick('scope'),
+      'aria-sort': selection === 'scope' ? direction : 'none',
+    },
+    {
+      children: (
+        <ButtonElement
+          variant="sort"
+          className={TABLE_HEADER_BUTTON_CLASS_NAME}
+        >
+          Type
+          <IconElement
+            variant={
+              selection === 'type' && direction !== 'none'
+                ? `sort-${direction}`
+                : 'sort-indeterminate'
+            }
+          />
+        </ButtonElement>
+      ),
+      key: 'th-type',
+      className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--type`,
+      onClick: () => onTableHeaderClick('type'),
+      'aria-sort': selection === 'type' ? direction : 'none',
+    },
+    {
+      children: (
+        <ButtonElement
+          variant="sort"
+          className={TABLE_HEADER_BUTTON_CLASS_NAME}
+        >
+          Permission
+          <IconElement
+            variant={
+              selection === 'permission' && direction !== 'none'
+                ? `sort-${direction}`
+                : 'sort-indeterminate'
+            }
+          />
+        </ButtonElement>
+      ),
+      key: 'th-permission',
+      className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--type`,
+      onClick: () => onTableHeaderClick('permission'),
+      'aria-sort': selection === 'permission' ? direction : 'none',
+    },
+  ];
+
+  const compareFn = LocationsViewColumnSortMap[selection];
+
+  if (direction === 'ascending') {
+    data.sort((a, b) => compareFn(a[selection], b[selection]));
+  } else {
+    data.sort((a, b) => compareFn(b[selection], a[selection]));
+  }
+
+  const rows = data.map((location, index) => [
+    {
+      key: `td-scope-${index}`,
+      children: (
+        <ButtonElement
+          className={TABLE_DATA_BUTTON_CLASS}
+          onClick={() => onLocationClick(location)}
+          variant="table-data"
+        >
+          {location.scope}
+        </ButtonElement>
+      ),
+    },
+    { key: `td-type-${index}`, children: location.type },
+    { key: `td-permission-${index}`, children: location.permission },
+  ]);
+
+  return { columns, rows };
+};
+
+export function LocationsViewTableControl(): React.JSX.Element | null {
```

Comment:
This can just be `DataTableControl` and `null` can be removed from the return type